packages/compiler/src/core/types.ts

```diff
@@ -2411,6 +2419,69 @@ export type LinterRuleDiagnosticReport<
   M extends keyof T = "default",
 > = LinterRuleDiagnosticReportWithoutTarget<T, M> & { target: DiagnosticTarget | typeof NoTarget };
```

Comment:
Am I right in thinking the only reason that linter has its own diagnostic types is because of `codefixes`, and that's not something we need for transformers?